contracts/Core/BlockManager.sol

```diff
@@ -194,23 +194,23 @@ contract BlockManager is Initializable, BlockStorage, StateManager, BlockManager
      * giveSorted() needs to be called to setup the dispute where in, the correct median will be calculated based on the votes
      * reported by stakers
      * @param epoch in which the dispute was setup and raised
-     * @param leafId index of the collection that is to be disputed
+     * @param collectionId index of the collection that is to be disputed
      * @param sortedValues values reported by staker for a collection in ascending order
      */
     function giveSorted(
         uint32 epoch,
-        uint16 leafId,
+        uint16 collectionId,
         uint256[] memory sortedValues
     ) external initialized checkEpochAndState(State.Dispute, epoch, buffer) {
-        require(leafId <= (collectionManager.getNumActiveCollections() - 1), "Invalid leafId");
-        uint256 medianWeight = voteManager.getTotalInfluenceRevealed(epoch, leafId) / 2;
+        require(collectionId <= (collectionManager.getNumActiveCollections()), "Invalid collectionId");
```

Comment:
@SamAg19 Suppose if the collectionIds are 1,2,3,4. Only collectionId 3 and 4 are active then `numActiveCollections` will be 2. Now when we try to pass collectionId as 3 which is actually correct then the contract will revert with the message "Invalid collectionId".

Can we change `collectionId <= (collectionManager.getNumActiveCollections())` to `collectionManager.getCollectionStatus(collectionId)` ?

Reply:
@SamAg19 this needs to be fixed.

Reply:
acknowledged
Fixed https://github.com/razor-network/oracle-contracts/pull/943/commits/0e97483a74c37258410e446c16d96fd2d658ecb6